winter/converters/converter.py

```diff
@@ -0,0 +1,153 @@
+import datetime
+import enum
+import inspect
+import typing
+
+import dataclasses
+from dateutil import parser
+
+from ..type_utils import get_origin_type
+from ..type_utils import is_optional
+
+_converters = {}
+
+
+class ConvertError(Exception):
+
+    def __init__(self, errors):
+        self.errors = errors
+
+
+def converter(type_: typing.Type, validator: typing.Callable = None):
+    def wrapper(func):
+        converters = _converters.setdefault(type_, [])
+        converters.append((func, validator))
+        return func
+
+    return wrapper
+
+
+def convert(data, hint_class):
+    origin_type = get_origin_type(hint_class)
+
+    types_ = origin_type.mro() if inspect.isclass(origin_type) else type(origin_type).mro()
+
+    for type_ in types_:
+        converters = _converters.get(type_, [])
+
+        for converter_, checker in converters:
+            if checker is None or checker(hint_class):
+                return converter_(data, hint_class)
+
+    raise ConvertError(f'Cannot convert "{data}"')
```

Comment:
Also, specify to what type it's failed to convert to.

Reply:
it is not so good to return to the front what name of class we use.
If we specify class there would be something like that:
 Cannot convert "{inlivid data}" to "winter.dataclasses.User"